src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,56 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int DATA_CAPACITY = 10;
+    private final Pair<K, V>[] data;
+    private int elementsCount;
+
+    public StorageImpl() {
+        this.data = new Pair[DATA_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = contains(key);
+        if (index > -1) {
+            data[index].value = value;
+            return;
+        }
+        data[elementsCount++] = new Pair<>(key, value);
     }
 
     @Override
     public V get(K key) {
-        return null;
+        int index = contains(key);
+        return index > -1 ? data[index].value : null;
     }
 
     @Override
     public int size() {
+        return elementsCount;
+    }
+
+    private int contains(K key) {
+        int index = 0;
+        for (Pair<K, V> pair : data) {
+            if (pair == null) {
+                break;
```

Comment:
Try to avoid `break` and `continue`. They are extremely complicated code.
```suggestion
                return -1;
```